CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -243,26 +243,48 @@ from UserCandies
             }
         }
 
-        public Candy TradeCandy(int candyID, int UserID)
+        public UserCandy GetCandyForUser(int candyID, int userID)
         {
-            var sql = @"select Candies.*
+            var sql = @"select UserCandies.UserCandyId, UserCandies.DateReceived
                         FROM Candies
                         JOIN UserCandies ON Candies.CandyId = UserCandies.CandyId
                         Where isConsumed = 0 
                         AND Candies.CandyId = @candyID
-                        AND UserCandies.UserId = @UserID;";
+                        AND UserCandies.UserId = @userID
+                        ORDER BY UserCandies.DateReceived;";
 
             using (var db = new SqlConnection(connectionString))
             {
                 var parameters = new
                 {
                     CandyId = candyID,
-                    UserId = UserID
+                    UserId = userID
                 };
 
-                var candyToBeTraded = db.QueryFirstOrDefault<Candy>(sql, parameters);
+                var candyToBeTraded = db.QueryFirstOrDefault<UserCandy>(sql, parameters);
                 return candyToBeTraded;
             }
         }
+        public bool UpdateTradedCandy(int receiverUserID, int userCandyIdToBeTraded)
+        {
+            // updates the usercandyid 
+            DateTime now = DateTime.Now;
+            var sql = @"update UserCandies
+                        set UserId = @receiverUserID
+                        where UserCandyId = @candyToBeTraded
+                        ";
+
+            using (var db = new SqlConnection(connectionString))
+            {
+                var parameters = new 
+                {
+                    receiverUserID = receiverUserID,
+                    candyToBeTraded = userCandyIdToBeTraded,
+                    DateReceived = now
```

Comment:
When planning we discussed we would leave the dateReceived as the original date that the inventory received the candy so the only item that needs to be updated is the id field to which user now owns the candy.

Reply:
Based on the sql query, it looks to me like we aren't actually changing the date, but we have some extra code for the date that isn't really being used and isn't needed.  I believe line 271 and line 283 could both be removed and this should still function as expected.  I don't have time to pull it down and test that right now, but I can do so this evening if no one has had a chance to test it before then